pkts-core/src/main/java/io/pkts/packet/IPPacket.java

```diff
@@ -1,273 +1,84 @@
-     * @param b
-     *            the second part of the IPv4 address, e.g. 168
-     * @param c
-     *            the third part of the IPv4 address, e.g. 0
-     * @param d
-     *            the fourth part of the IPv4 address, e.g. 100
-     */
-    void setDestinationIP(int a, int b, int c, int d);
-
-    void setDestinationIP(byte a, byte b, byte c, byte d);
-
-    /**
-     * Set the destination IP of this {@link IPPacket}. Note, using
-     * {@link #setDestinationIP(int, int, int, int)} will be must faster so try
-     * and use it instead.
-     * 
-     * @param sourceIp
-     */
-    void setDestinationIP(String destinationIP);
-
-    /**
-     * This 16-bit field defines the entire packet (fragment) size, including
-     * header and data, in bytes. The minimum-length packet is 20 bytes (20-byte
-     * header + 0 bytes data) and the maximum is 65,535 bytes — the maximum
-     * value of a 16-bit word. The largest datagram that any host is required to
-     * be able to reassemble is 576 bytes, but most modern hosts handle much
-     * larger packets. Sometimes subnetworks impose further restrictions on the
-     * packet size, in which case datagrams must be fragmented. Fragmentation is
-     * handled in either the host or router in IPv4.
-     * 
-     * (source: http://en.wikipedia.org/wiki/IPv4)
-     * 
-     * @return
-     */
-    int getTotalIPLength();
-
-    /**
-     * The checksum of the IP-packet. The checksum in an IP-packet is a 16 bit
-     * checksum of the header bytes (which the checksum set to zero) and is
-     * returned as a unsigned short (hence an int)
-     * 
-     * Checkout
-     * 
-     * @return
-     */
-    int getIpChecksum();
-
-    /**
-     * After you change anything in an IP packet (apart from the payload) you
-     * should re-calculate the checksum. If you don't, if this then is written
-     * to a pcap and later opened in e.g. wireshark, then all packets will be
-     * flagged as bad checksums.
-     */
-    void reCalculateChecksum();
-
-    boolean verifyIpChecksum();
-
-    @Override
-    IPPacket clone();
-
-    /**
-     * The IP version (4 or 6)
-     * 
-     * @return
-     */
-    int getVersion();
-
-    /**
-     * Get the length of the IP headers (in bytes)
-     * 
-     * @return
-     */
-    int getHeaderLength();
-
-    /**
-     * Note, this should be treated as a unsigned short.
-     * 
-     * This field is an identification field and is primarily used for uniquely
-     * identifying fragments of an original IP datagram. Some experimental work
-     * has suggested using the ID field for other purposes, such as for adding
-     * packet-tracing information to help trace datagrams with spoofed source
-     * addresses
-     * 
-     * (source http://en.wikipedia.org/wiki/IPv4)
-     * 
-     * @return
-     */
-    int getIdentification();
-
-    /**
-     * 
-     * @return
-     */
-    boolean isFragmented();
```

Comment:
I feel that perhaps this one should be left on the IPPacket interface since all ip packets can be fragmented.